diff/reader_util.go

```diff
@@ -25,6 +24,14 @@ func readLine(r *bufio.Reader) ([]byte, error) {
 	} else if err != nil {
 		return nil, err
 	}
-	line := bytes.TrimSuffix(line_, []byte{'\n'})
-	return line, nil
+	line := line_[0:len(line_)-1]
```

Comment:
After successfully reading a line we can guarantee that `line_` has a trailing `'\n'`